simcon_project/static/conversation_templates/create_conversation_template.js

```diff
@@ -0,0 +1,346 @@
+    }
+
+    nodes.forEach((value, key) => {
+        if(value !== currentNodeInFocus) {
+            if(selectedValue == key) {
+                literalToReturn += `<option selected value="${key}">${nodes.get(key).nodeName}</option>`
+            } else {
+                literalToReturn += `<option value="${key}">${nodes.get(key).nodeName}</option>`
+            }
+        }
+    })
+    return literalToReturn
+}
+
+/**
+ * Used to delete a choice (both from the choice card column and current node's choices map)
+ * @param choiceIndex
+ */
+function removeChoice(choiceIndex) {
+    let choiceCard = document.getElementById("choice-" + choiceIndex)
+    choiceCard.parentElement.removeChild(choiceCard)
+    currentNodeInFocus.responseChoices.delete(choiceIndex)
+}
+
+/**
+ *
+ * @param url YouTube url to embed. Does not have to match embeddable style
+ * @returns {string}
+ */
+function getEmbeddableUrl(url) {
+    REGEX = /^.*((youtu.be\/)|(v\/)|(\/u\/\w\/)|(embed\/)|(watch\?))\??v?=?([^#&?]*).*/
+    let match = url.match(REGEX)
+    regexedUrl = (match&&match[7].length==11)? match[7] : ""
+    if(!(regexedUrl == "")) {
+        return "https://www.youtube.com/embed/" + regexedUrl
+    }
+    return ""
+}
+
+// Here are all the DOM getters/setters... technically we don't need standalone functions since most are only used once
+// However, we might end up needing them once we are setting up template edit functionality... if not, they can always be moved inline
+
+function getTemplateNameInput() {
+    return document.getElementById('template-name-input').value
+}
+
+function getTemplateDescriptionInput() {
+    return document.getElementById('template-description-input').value
+}
+
+function getNodeNameInput() {
+    return document.getElementById('node-name-input').value
+}
+
+function setNodeNameInput(name) {
+    document.getElementById('node-name-input').value = name
+}
+
+function getNodeDescriptionInput() {
+    return document.getElementById('node-description-input').value
+}
+
+function setNodeDescriptionInput(description) {
+    document.getElementById('node-description-input').value = description
+}
+
+function getFirstStepToggle() {
+    return document.getElementById('is-first-node-check').checked
+}
+
+function setFirstStepToggle(checked) {
+    document.getElementById('is-first-node-check').checked = checked
+}
+
+function getTerminalToggle() {
+    return document.getElementById('is-terminal-node-check').checked
+}
+
+function setTerminalToggle(checked) {
+    document.getElementById('is-terminal-node-check').checked = checked
+}
+
+function getVideoUrlInput() {
+    return document.getElementById('video-url-input').value
+}
+
+function setVideoUrlInput(url) {
+    document.getElementById('video-url-input').value = url
+}
+
+function setEmbeddedVideoUrl(url) {
+    document.getElementById("embedded-video-iframe").setAttribute('src', getEmbeddableUrl(url))
+}
```

Comment:
How much of this is from the old simcon? Lol. If this is all you, then hot diggity dog...

Reply:
Thanks haha... wish I could do it the Django way tho